File: abc357/a.py
def main():
    N, M = map(int, input().split())
    H = list(map(int, input().split()))
    sum = 0
    for i in range(N):
        sum += H[i]
        if sum > M:
            print(i)
            return
    print(i+1)
    return

main()

File: abc357/a.ts
function a375(inputs:string){
    const [NM, Hs]: string[] = inputs.trim().split("\n");
    const [N, M]: number[] = NM.split(" ").map(Number);
    const numberH: number[] = Hs.split(" ").map(Number);
    let sum: number = 0;
    let result: number = 0;
    let i: number = 0;
    do {
        sum += numberH[i];
        result++;
        i++;
    } while (sum <= M);

    console.log(result - 1);
}   
a375(require("fs").readFileSync("/dev/stdin", "utf8"));